styx-common/src/main/java/com/spotify/styx/util/BasicWorkflowValidator.java

```diff
@@ -134,6 +137,15 @@ public BasicWorkflowValidator(DockerImageValidator dockerImageValidator) {
       }
     });
 
+    cfg.retryCondition().ifPresent(retryCondition -> {
+      upperLimit(e, retryCondition.length(), MAX_RETRY_CONDITION_LENGTH, "retry condition too long");
```

Comment:
If this `upperLimit` check generates an error, the rest of the body will run and the expression will still be parsed. I think we should skip parsing it in that case. Since it might be any arbitrary length I think we should avoid doing the effort of parsing it.

Reply:
This keeps the same behaviour as any other validation. We always try to find as many problems as possible.

Reply:
I understand, however I see this piece of config to be of different nature. Some checks make sense to have a level of priority of others, such as this max length. I see the max length restriction to have the purpose of saving Styx cpu cycles and avoiding too long parsing times and users going crazy with their conditions. If the check fails, but then we still parse the expression - we've lost the point of the restriction.

Reply:
I see this as what a normal compiler would always do: parsing as much as possible and finding all problems. And this is the same as what we have been doing to service account validation: first check length then check regex.  Parsing the expression is relatively fast (not much slower than regex validation). This piece of code is also used by styx cli so it is not needed to save CPU cycles there.

Reply:
To me another point to think here is: when later on we need to validate another thing, should we stop earlier or not? Once opening an example, every time we need to validate different aspects we will be trapped into this kind of struggling.

Reply:
Alright, makes sense